components/fxa-client/src/migrator.rs

```diff
@@ -24,20 +31,44 @@ impl FirefoxAccount {
         k_sync: &str,
         k_xcs: &str,
         copy_session_token: bool,
-    ) -> Result<()> {
+    ) -> Result<FxAMigrationResult> {
         // if there is already a session token on account, we error out.
         if self.state.session_token.is_some() {
             return Err(ErrorKind::IllegalState("Session Token is already set.").into());
         }
 
-        let migration_session_token = if copy_session_token {
+        self.state.in_flight_migration = Some(MigrationData {
+            k_sync: k_sync.to_string(),
+            k_xcs: k_xcs.to_string(),
+            copy_session_token,
+            session_token: session_token.to_string(),
+        });
+
+        self.try_migration()
+    }
+
+    /// Check if the client is
+    pub fn is_in_migration_state(&self) -> bool {
+        self.state.in_flight_migration.is_some()
+    }
+
+    pub fn try_migration(&mut self) -> Result<FxAMigrationResult> {
```

Comment:
Naming nit: I think the overall structure of the code would be a bit clearer if this were called `retry_migrate_from_session_token` for symmetry with what it's called in the higher layers.